tests/test_mask.py

```diff
@@ -244,7 +244,7 @@ def test_mask_pad(basic_image_2x2, basic_image_file, basic_geometry):
 
     geometries = [basic_geometry]
     with rasterio.open(basic_image_file) as src:
-        masked, transform = mask(src, geometries, crop=True, pad=True)
```

Comment:
@derekjanni I'd like to keep this test as it is. Will you please add another test function that checks that `pad=4` does what you expect?

Reply:
Sure. I added a check for padding of 2px and 4px. The 4 case isn't really a good test on its own, because with 4px of padding, the input 10x10 image is all within the mask, however, it's a good edge case to check that things don't break when that number goes up past the frames of the input image.